x/perp/v2/keeper/keeper.go

```diff
@@ -33,7 +33,9 @@ type Keeper struct {
 	Positions        collections.Map[collections.Pair[asset.Pair, sdk.AccAddress], types.Position]
 	ReserveSnapshots collections.Map[collections.Pair[asset.Pair, time.Time], types.ReserveSnapshot]
 	DnREpoch         collections.Item[uint64]
-	TraderVolumes    collections.Map[collections.Pair[sdk.AccAddress, uint64], math.Int] // Keeps track of user volumes for each epoch.
+	TraderVolumes    collections.Map[collections.Pair[sdk.AccAddress, uint64], math.Int]         // Keeps track of user volumes for each epoch.
+	GlobalDiscounts  collections.Map[math.Int, math.LegacyDec]                                   // maps a volume level to a discount
+	TraderDiscounts  collections.Map[collections.Pair[sdk.AccAddress, math.Int], math.LegacyDec] // maps a user and volume level to a discount, supersedes global discounts
```

Comment:
As a trader may have a discount set independent of their volume, you may want to combine the `TradeVolumeLastEpoch` and `TradeDiscount` state together into one struct.  

This would let you use the proto encoder rather than a custom `sdk.Int` one for volume. And since `TraderVolumes` and `TraderDiscounts` are both trader specific, we could just group them as:

`Traders collections.Map[...]` ?

Just thinking of how to optimize the schema a bit to save some work.